iron-fit-behavior.html

```diff
@@ -600,23 +639,24 @@
           continue;
         }
 
-        position = position || pos;
         pos.croppedArea = this.__getCroppedArea(pos, size, fitRect);
-        var diff = pos.croppedArea - position.croppedArea;
-        // Check which crops less. If it crops equally, check if align is ok.
-        if (diff < 0 || (diff === 0 && alignOk)) {
-          position = pos;
-        }
         // If not cropped and respects the align requirements, keep it.
         // This allows to prefer positions overlapping horizontally over the
         // ones overlapping vertically.
-        if (position.croppedArea === 0 && alignOk) {
+        if (pos.croppedArea === 0 && alignOk) {
+          position = pos;
           break;
         }
+        position = position || pos;
+        var diff = pos.croppedArea - position.croppedArea;
+        // Check which crops less. If it crops equally, check if at least one align setting is ok.
+        if (diff < 0 || (diff === 0 && (pos.verticalAlign === vAlign || pos.horizontalAlign === hAlign))) {
+          position = pos;
+        }
```

Comment:
Why were these lines moved below the if statement? Does this fix another issue?

Reply:
Mainly for clarity: we need to compute the diff cropped area only if we didn't find the "perfect match". The difference from the previous implementation is that we don't check for both vertical and horizontal alignment to match, but rather one or the other.